node/malus/src/malus.rs

```diff
@@ -68,8 +68,11 @@ impl MalusCli {
 		match self.variant {
 			NemesisVariant::BackGarbageCandidate(cmd) =>
 				polkadot_cli::run_node(run_cmd(cmd), BackGarbageCandidate, finality_delay)?,
-			NemesisVariant::SuggestGarbageCandidate(cmd) =>
-				polkadot_cli::run_node(run_cmd(cmd), BackGarbageCandidateWrapper, finality_delay)?,
+			NemesisVariant::SuggestGarbageCandidate(opts) => polkadot_cli::run_node(
+				run_cmd(opts.clone().cmd),
+				BackGarbageCandidateWrapper::new(opts),
+				finality_delay,
+			)?,
 			NemesisVariant::DisputeAncestor(opts) => polkadot_cli::run_node(
```

Comment:
Don't we also want to implement the same behaviour here ?

Reply:
Regarding the `DisputeAncestor`, the logic is conditional, so can't add probabilities in there since it is the `SuggestGarbageCandidate` command which is responsible for creating bad candidates.
The `Option` structs are passed in the `NemesisVariant` `enum`. For example: `SuggestGarbageCandidate(SuggestGarbageCandidateOptions)` has the `SuggestGarbageCandidateOption` struct as the `opts`. So, because `DisputeAncestor` doesn't need the percentage parameter, we do not need to implement the same probabilistic behavior there as far as I know.

Reply:
@sandreim added the 100% for `suggest-garbage-candidate` variant, and the CLI parameter for the other two variants.